app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+
+
+
+/**
+ * Class that contains the methods to read the allergens from a file, create an allergen list,
+ * find the allergens in the recognized ingredients from the OCR.
+ */
+class AllergensManager {
+
+    private final int COMMON_NAME = 0;
+    private final int INCI_NAMES = 1;
+
```

Comment:
I couldn't understand the usage of these constants until I have seen them in usage below, so I would suggest to give them more specific names, such as "COMMON_NAME_CSV_INDEX" and "INCI_NAMES_CSV_INDEX". May it be a sensible change in your opinion?

Reply:
Maybe it would be better to just add a comment explaining what they are, but you are right.

Reply:
Yes, your is definitely an easier and more effective solution